RTCCodecStats - add docs (#32452)
mdn/content: files/en-us/web/api/rtccodecstats/timestamp/index.md

---
title: "RTCCodecStats: timestamp property"
short-title: timestamp
slug: Web/API/RTCCodecStats/timestamp
page-type: web-api-instance-property
browser-compat: api.RTCStatsReport.type_codec.timestamp
---

{{APIRef("WebRTC")}}

The **`timestamp`** property of the {{domxref("RTCCodecStats")}} dictionary is a {{domxref("DOMHighResTimeStamp")}} object specifying the time at which the data in the object was sampled.

## Value

> A {{domxref("DOMHighResTimeStamp")}} value indicating the time at which the activity described by the statistics in this object was recorded, in milliseconds elapsed since the beginning of January 1, 1970, UTC.

Reviewer: ```suggestion
A {{domxref("DOMHighResTimeStamp")}} value indicating the time when the activity described by the statistics in this object was recorded. It is measured in milliseconds since January 1, 1970, UTC.
```

Author: I'm good with this but have added to my notes to change globally once this and my other ones merge.

> The value should be accurate to within a few milliseconds but may not be entirely precise, either because of hardware or operating system limitations or because of [fingerprinting](/en-US/docs/Glossary/Fingerprinting) protection in the form of reduced clock precision or accuracy.

Reviewer: Is this what we intend to say here:
```suggestion
The value is accurate to within a few milliseconds, though it might not be entirely precise because of hardware or operating system limitations. Additionally, [fingerprinting](/en-US/docs/Glossary/Fingerprinting) protection could reduce the precision or accuracy of the time reported.
```

Author: No. My versions says "should be" within a few ms, whereas your version must be accurate to within a few ms irrespective of all factors. I actually don't know which it is, so would rather say "should be".

The bigger problem is that this text is many years old - all I've done is propagated the original info in a dictionary. I can't be sure from the spec that this is still true. I've asked here https://github.com/w3c/webrtc-stats/issues/786

Reviewer: Got it - "is" vs "should be", makes sense

Good move to to seek clarification 👍
